### DISCUSSION TITLE
TransformScene | How to write the transform matrix?

### DISCUSSION THREAD

**Discussion by ghost** [cdcseacave/openMVS#1022] [Q&A] [answered]
How is the transform matrix written?

**Comment by ghost** [comment, marked as answer]
1
    0
    0
    0
    0
    1
    0
    0
    0
    0
    1
    1
or

    1
    0
    0
    0
    0
    1
    0
    0
    0
    0
    1
    0
    0
    0
    0
    1

**Reply by ghost** [reply]
Thanks me!

**Comment by cdcseacave** [comment]
Hey, sry for the lack of documentation, but indeed the transformation matrix can be either 3x4 or 4x4, and row-major format
Can you pls contact me on my email? I see you are using a lot OpenMVS and I am curious to find out more of what you do.

**Reply by Jansschen** [reply]
How to use the TransformScene?

**Reply by Jansschen** [reply]
21:29:53 [App     ] OpenMVS x32 v2.2.0
21:29:53 [App     ] Build date: Jul  5 2023, 19:23:03
21:29:53 [App     ] CPU:  (8 cores)
21:29:53 [App     ] RAM: 16.00GB Physical Memory 16.00GB Virtual Memory
21:29:53 [App     ] OS: Darwin 23.0.0 (arm64)
21:29:53 [App     ] Disk: 
21:29:53 [App     ] warning: no SSE compatible CPU or OS detected
21:29:53 [App     ] Command line: TransformScene ./scene.mvs
21:29:53 [App     ] Available options:

Generic options:
  -h [ --help ]                         produce this help message
  -w [ --working-folder ] arg           working directory (default current 
                                        directory)
  -c [ --config-file ] arg (=TransformScene.cfg)
                                        file name containing program options
  --export-type arg (=ply)              file type used to export the 3D scene 
                                        (ply, obj, glb or gltf)
  --archive-type arg (=4294967295)      project archive type: 0-text, 1-binary,
                                        2-compressed binary
  --process-priority arg (=-1)          process priority (below normal by 
                                        default)
  --max-threads arg (=0)                maximum number of threads (0 for using 
                                        all available cores)
  -v [ --verbosity ] arg (=2)           verbosity level

Main options:
  -i [ --input-file ] arg               input scene filename
  -o [ --output-file ] arg              output filename for storing the scene
  -a [ --align-file ] arg               input scene filename to which the scene
                                        will be cameras aligned
  -t [ --transform-file ] arg           input transform filename by which the 
                                        scene will transformed
  --transfer-texture-file arg           input mesh filename to which the 
                                        texture of the scene's mesh will be 
                                        transfered to (the two meshes should be
                                        aligned and the new mesh to have 
                                        UV-map)
  --indices-file arg                    input indices filename to be used with 
                                        ex. texture transfer to select a subset
                                        of the scene's mesh
  --compute-volume arg (=0)             compute the volume of the given 
                                        watertight mesh, or else try to 
                                        estimate the ground plane and assume 
                                        the mesh is bounded by it
  --plane-threshold arg (=0)            threshold used to estimate the ground 
                                        plane (<0 - disabled, 0 - auto, >0 - 
                                        desired threshold)
  --sample-mesh arg (=-300000)          uniformly samples points on a mesh (0 -
                                        disabled, <0 - number of points, >0 - 
                                        sample density per square unit)
  --up-axis arg (=2)                    scene axis considered to point upwards 
                                        (0 - x, 1 - y, 2 - z

**Reply by Jansschen** [reply]
i try insert this command "sudo ./RefineMesh ./scene_mesh.mvs --max-face-area 16"
